python/pipeline/odor.py

```diff
@@ -265,3 +267,106 @@ class MesoMatch(dj.Manual):
     ---
     -> meso.ScanInfo
     """
+
+
+@schema
+class OdorAnalysis(dj.Computed):
+    definition = """
+    -> experiment.ExperimentalIdentifier
+    """
+
+    class CombinedTrial(dj.Part):
+        definition = """
+        -> master
+        trial_idx               : smallint unsigned    # trial index for recording
```

Comment:
the syntax `-> OrdorTrials(trial_idx)` doesn't work?
maybe try `(trial_idx) -> OrdorTrials(trial_idx)`

Reply:
`(trial_idx) -> OdorTrials(trial_idx)` worked, thanks.

unfortunately, `-> OdorTrials(trial_idx)` had the following error:

------------------------------------------------------------------------
NameError                              Traceback (most recent call last)
/usr/local/lib/python3.6/dist-packages/datajoint/declare.py in compile_foreign_key(line, context, attributes, primary_key, attr_sql, foreign_key_sql, index_sql)
    141     try:
--> 142         ref = eval(result.ref_table, context)
    143     except NameError if obsolete else Exception:

/data/odor_meso/python/pipeline/odor.py in <module>

NameError: name 'trial_idx' is not defined

During handling of the above exception, another exception occurred:

DataJointError                         Traceback (most recent call last)
<ipython-input-1-6f93fb2c4bf5> in <module>
----> 1 from pipeline import odor

/data/odor_meso/python/pipeline/odor.py in <module>
    271 
    272 @schema
--> 273 class OdorAnalysis(dj.Computed):
    274     definition = """
    275     -> experiment.ExperimentalIdentifier

/usr/local/lib/python3.6/dist-packages/datajoint/schemas.py in __call__(self, cls, context)
    225                     # allow addressing master by name or keyword 'master'
    226                     self.process_table_class(part, context=dict(
--> 227                         context, master=cls, self=part, **{cls.__name__: cls}))
    228         return cls
    229 

/usr/local/lib/python3.6/dist-packages/datajoint/schemas.py in process_table_class(self, table_class, context, assert_declared)
    188                 raise DataJointError('Table `%s` not declared' % instance.table_name)
    189             else:
--> 190                 instance.declare(context)
    191         is_declared = is_declared or instance.is_declared
    192 

/usr/local/lib/python3.6/dist-packages/datajoint/table.py in declare(self, context)
     59             raise DataJointError('Cannot declare new tables inside a transaction, '
     60                                  'e.g. from inside a populate/make call')
---> 61         sql, external_stores = declare(self.full_table_name, self.definition, context)
     62         sql = sql.format(database=self.database)
     63         try:

/usr/local/lib/python3.6/dist-packages/datajoint/declare.py in declare(full_table_name, definition, context)
    279 
    280     table_comment, primary_key, attribute_sql, foreign_key_sql, index_sql, external_stores = prepare_declare(
--> 281         definition, context)
    282 
    283     if not primary_key:

/usr/local/lib/python3.6/dist-packages/datajoint/declare.py in prepare_declare(definition, context)
    247             compile_foreign_key(line, context, attributes,
    248                                 primary_key if in_key else None,
--> 249                                 attribute_sql, foreign_key_sql, index_sql)
    250         elif re.match(r'^(unique\s+)?index[^:]*$', line, re.I):   # index
    251             compile_index(line, index_sql)

/usr/local/lib/python3.6/dist-packages/datajoint/declare.py in compile_foreign_key(line, context, attributes, primary_key, attr_sql, foreign_key_sql, index_sql)
    142         ref = eval(result.ref_table, context)
    143     except NameError if obsolete else Exception:
--> 144         raise DataJointError('Foreign key reference %s could not be resolved' % result.ref_table)
    145 
    146     options = [opt.upper() for opt in result.options]

DataJointError: Foreign key reference OdorTrials(trial_idx) could not be resolved